### PR TITLE
AUTO-878 Automate the curation of generated ELK indices older than 30 days

### PR DESCRIPTION
**Pull request by snayler0**

### CODE REVIEW

File: monitoring/deploy_eck_logging.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+from __future__ import print_function
```
Comment (ga-autobots) [inline monitoring/deploy_eck_logging.py]: remove

File: monitoring/deploy_eck_logging.py [hunk 2]
```diff
@@ -472,13 +533,21 @@ def main():
                         help='What name to give the elk instance. default: elk')
     parser.add_argument('-a', '--action',
                         default='create',
-                        help='The action to perform. options: create, delete, or clean. Delete will delete all elk '
-                             'objects with the provided name (-n). Clean will delete indexes older than 30 days from '
-                             'the elastic search domain name (-n) provided. default: create')
+                        help='The action to perform. options: create, or delete. Delete will delete all elk '
+                             'objects with the provided name (-n). default: create')
     parser.add_argument('-c', '--cidr',
```
Comment (ga-autobots) [inline monitoring/deploy_eck_logging.py]: default 0.0.0.0/0

File: monitoring/deploy_eck_logging.py [hunk 3]
```diff
@@ -336,47 +345,74 @@ def update_lambda_permissions(lambda_arn, boto_session):
     )
 
 
-def run_curator(name, boto_session):
+def create_curator_lambda(domainname, role_arn):
     """
-    Cleans out any indexes older than 30 days.
-
+    Uploads this script into lambda and sets up a scheduled cloud watch rule to clean up elastic search indices that are
```
Comment (ga-autobots) [inline monitoring/deploy_eck_logging.py]: uploads eck_curator.py

File: monitoring/eck_curator.py [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    headers = dict(request.headers.items())
+
+    opener = urllib2.build_opener(urllib2.HTTPHandler)
+    request = urllib2.Request(url)
+    request.add_header('X-Amz-Date', headers['X-Amz-Date'])
+    request.add_header('X-Amz-Security-Token', headers['X-Amz-Security-Token'])
+    request.add_header('Authorization', headers['Authorization'])
+    request.get_method = lambda: method
+
+    return opener.open(request).read()
```
Comment (ga-autobots) [inline monitoring/eck_curator.py]: AWSRequest.read()?

File: monitoring/eck_curator.py [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+    credentials = boto3.session.Session().get_credentials()
+
+    url = 'https://{0}{1}'.format(endpoint, path)
+
+    request = AWSRequest(method=method, url=url)
+    SigV4Auth(credentials, service, region).add_auth(request)
+
+    headers = dict(request.headers.items())
+
+    opener = urllib2.build_opener(urllib2.HTTPHandler)
```
Comment (ga-autobots) [inline monitoring/eck_curator.py]: Make this into function that accepts path, method?
AWSRequest.read()?

**Comment by aynsof** [comment]
👍 

[Approved with PullApprove](https://pullapprove.com/GeoscienceAustralia/toucan/pull-request/25/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=GeoscienceAustralia/toucan)

**Pull request merged**